### PR TITLE
Introduce pool client idle timeout in session pooling mode

### PR DESCRIPTION
**Pull request by reshke**
```current time 12868806711, drop horizon 12828797602
1155 2021-02-11T19:06:05Z info [cf80ef47cc411 s4ec75657858f] (shutdown) drop client idle connection on due timeout 20 sec
```

### CODE REVIEW

File: sources/frontend.c [hunk 1]
```diff
@@ -474,27 +474,77 @@ static inline bool od_eject_conn_with_rate(od_client_t *client,
...
 
-static inline bool od_should_drop_connection(od_client_t *client,
-					     od_server_t *server)
+static inline bool od_eject_conn_with_timeout(od_client_t *client,
+					      od_server_t *server,
+					      od_instance_t *instance,
+					      int timeout)
 {
-	od_instance_t *instance = client->global->instance;
+	const uint64_t interval_usec = 1000000;
```
Comment (x4m) [inline sources/frontend.c:482]: Can we have a timeout in microseconds instead of multiplying it here and there?

File: sources/frontend.c [hunk 2]
```diff
@@ -474,27 +474,77 @@ static inline bool od_eject_conn_with_rate(od_client_t *client,
...
 
-	//TODO:: drop no more than X connection per sec/min/whatever
-	if (od_likely(instance->shutdown_worker_id == INVALID_ID)) {
-		// try to optimize likely path
-		return false;
+	if (server == NULL) {
```
Comment (x4m) [inline sources/frontend.c:484]: Are we considering any client without server expired independently from timeout?

File: sources/frontend.c [hunk 3]
```diff
@@ -474,27 +474,77 @@ static inline bool od_eject_conn_with_rate(od_client_t *client,
...
+				    od_server_synchronized(server))) {
+				return od_eject_conn_with_timeout(
+					client, server, instance,
+					client->rule->pool_client_idle_timeout);
+			}
+		}
+	}
 		/* fall through */
 	case OD_RULE_POOL_TRANSACTION: {
+		//TODO:: drop no more than X connection per sec/min/whatever
```
Comment (x4m) [inline sources/frontend.c:536]: So long-running transactions now survive pool_client_idle_timeout. Is it expected?

**Review by x4m — CHANGES_REQUESTED** [review]

File: sources/frontend.c (hunk 3 above)
Comment (reshke) [reply sources/frontend.c:549]: separate setting, pool idle in transaction timeout

**Review by x4m — APPROVED** [review]
LGTM

**Pull request merged**